codec/src/main/java/io/netty/handler/codec/ByteToMessageDecoder.java

```diff
@@ -265,8 +265,8 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
                 callDecode(ctx, cumulation, out);
             } catch (DecoderException e) {
                 throw e;
-            } catch (Throwable t) {
-                throw new DecoderException(t);
+            } catch (Exception e) {
+                throw new DecoderException(e);
```

Comment:
optional: maybe don't wrap RuntimeExceptions either?

Reply:
I think we should just handle `Error` special here imho